uni/lib/view/home/widgets/main_cards_list.dart

```diff
@@ -211,16 +245,16 @@ class MainCardsListState extends State<MainCardsList> {
   }
 
   void removeCardIndexFromFavorites(int i, BuildContext context) {
-    final favorites = List<FavoriteWidgetType>.from(widget.favoriteCardTypes)
-      ..removeAt(i);
-    widget.saveFavoriteCards(favorites);
+    setState(() {
+      widget.favoriteCardTypes.removeAt(i);
+    });
   }
 
   void addCardToFavorites(FavoriteWidgetType type, BuildContext context) {
-    final favorites = List<FavoriteWidgetType>.from(widget.favoriteCardTypes);
-    if (!favorites.contains(type)) {
-      favorites.add(type);
-    }
-    widget.saveFavoriteCards(favorites);
+    setState(() {
+      if (!widget.favoriteCardTypes.contains(type)) {
+        widget.favoriteCardTypes.add(type);
+      }
+    });
```

Comment:
A widget's props are meant to be unmodifiable. So this main card list receives a list of favorite cards and should limit itself to showing it. If it needs to change it, a callback should be invoked to change state on the parent. Here, it is `saveFavoriteCards`.
What you are trying to achieve is not reset editing mode after invoking that callback. What perhaps needs to be changed is notify the parent that it needs to build this in editing mode. A (uglier) alternative is to make the bool here static.

Reply:
Huuum, okok, that explanation makes sense. Thank you for clarifying :)